### ISSUE TITLE
what about asyncio.gather?

### ISSUE THREAD

**Issue by wedobetter** [mjpieters/aiolimiter#69]
Unless I am doing something wrong, it wont work with that.
My code looks like this:

`
async def fetch(something):
    await get(something)

async with limiter:
  await asyncio.gather(*[fetch(t) for t in range(100)])
`

Can you provide an example?

Thank you

**Comment by mjpieters** [comment]
You are gating access to the `asyncio.gather()` call. You haven't made it explicit, but I'm going to assume that that's not what you wanted, but that you instead wanted to limit `get()` calls.

Gather schedules each entry as a separate task, so they run in parallel **independently**. You are rate limiting access to the `asyncio.gather()` call. If you want to rate limit the `get()` calls, put your `with` block _there_, in the task execution.

You can share the limiter between tasks; you could pass it in as an argument:

```python
async def fetch(limiter, something):
    async with limiter:
        await get(something)

 await asyncio.gather(*[fetch(limiter, t) for t in range(100)])
```

So when each task runs, *each task* has to first ask for the rate limiter if it can run the `get()` call.
